src/inmanta/data/__init__.py

```diff
@@ -4072,7 +4072,12 @@ def __init__(self, from_postgres: bool = False, **kwargs: object) -> None:
         if from_postgres and self.messages:
             new_messages = []
             for message in self.messages:
-                message = json.loads(message)
+                # Not 100% sure why this is needed, could depend on whether the data is
+                # retrieved through sql alchemy (automatically deserializes json into dict)
+                # or through regular asyncpg queries (no automatic deserialization)
+                if isinstance(message, str):
+                    message = json.loads(message)
+                assert isinstance(message, dict)
```

Comment:
Still need to address this

Reply:
SQl alchemy does some reconfiguration of asyncpg, but I don't get why it would be selective? I would expect it to always be either the one or the other?

Reply:
and if you keep the other branch? (I would expect to always get a dict when using sqlalchemy?)